cli/azd/pkg/cmdsubst/containerAppImage.go

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package cmdsubst
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/containerapps"
+	"github.com/azure/azure-dev/cli/azd/pkg/infra"
+	"github.com/azure/azure-dev/cli/azd/pkg/project"
+)
+
+// ContainerAppImageCommandName is the name of the command that can be used to fetch the current image for a container in
+// a container app. It expects two arguments, the first is the name of a service in azure.yaml and the second is the name
+// of the container in the template for that container app you want to pull the image id from.
+//
+// If the container app has not yet been created, containerAppImage returns an empty string.
+//
+// The expected use of this command is in main.parameters.json files, like this:
+//
+//	"imageName": {
+//	  "value": "${SERVICE_WEB_IMAGE_NAME:=$(containerAppImage web main)}"
+//	}
+//
+// Here, we use the value of SERVICE_WEB_IMAGE_NAME if it is set (this value is updated in the .env file every time
+// `azd deploy` is run) and if it is not set (for example, in CI where this value is not persisted across runs), the
+// current value is pulled from the live Container App resource.
+//
+// This can be useful to ensure that the container app image is not reset to an initial value.
+const ContainerAppImageCommandName string = "containerAppImage"
+
+type ContainerAppImageCommandExecutor struct {
+	resourceManager      project.ResourceManager
+	containerAppsService containerapps.ContainerAppService
+	subscriptionId       string
+	services             map[string]*project.ServiceConfig
+}
```

Comment:
I'm not a fan on creating these custom functions on top of bicep.  It makes the params file less portable and not really native bicep.  I'd much rather just have some env var that they can reference and use it if they need it.